extensions/aws-ai/src/main/java/ai/djl/aws/sagemaker/SageMaker.java

```diff
@@ -0,0 +1,670 @@
+
+/** A utility class that help deploy model to SageMaker. */
+public final class SageMaker {
+
+    private static final Logger logger = LoggerFactory.getLogger(SageMaker.class);
+    private static final char[] CHARS = "abcdefghijklmnopqrstuvwxyz1234567890.-".toCharArray();
+
+    private SageMakerClient sageMaker;
+    private SageMakerRuntimeClient smRuntime;
+    private S3Client s3;
+    private IamClient iam;
+    private Region region;
+    private Model model;
+    private String modelName;
+    private String bucketName;
+    private String bucketPath;
+    private String executionRole;
+    private String containerImage;
+    private String endpointConfigName;
+    private String endpointName;
+    private String instanceType;
+    private int instanceCount;
+
+    private SageMaker(Builder builder) {
+        sageMaker = builder.sageMaker;
+        smRuntime = builder.smRuntime;
+        s3 = builder.s3;
+        iam = builder.iam;
+        model = builder.model;
+        modelName = model.getName();
+        bucketName = builder.bucketName;
+        bucketPath = builder.bucketPath;
+        executionRole = builder.executionRole;
+        containerImage = builder.containerImage;
+        endpointConfigName = builder.endpointConfigName;
+        endpointName = builder.endpointName;
+        instanceType = builder.instanceType;
+        instanceCount = builder.instanceCount;
+        region = DefaultAwsRegionProviderChain.builder().build().getRegion();
+    }
+
+    /**
+     * Deploys the model to Amazon SageMaker hosting service.
+     *
+     * @throws IOException if failed upload model to S3
+     */
+    public void deploy() throws IOException {
+        DescribeEndpointRequest describeReq =
+                DescribeEndpointRequest.builder().endpointName(endpointName).build();
+        SageMakerWaiter waiter = sageMaker.waiter();
+        if (doesEndpointExist()) {
+            throw new IllegalStateException("Endpoint already exists: " + endpointName);
+        }
+        createEndpointConfig();
+
+        logger.info("Creating endpoint {} ...", endpointName);
+        CreateEndpointRequest req =
+                CreateEndpointRequest.builder()
+                        .endpointName(endpointName)
+                        .endpointConfigName(endpointConfigName)
+                        .build();
+        CreateEndpointResponse resp = sageMaker.createEndpoint(req);
+        String endpointArn = resp.endpointArn();
+        waiter.waitUntilEndpointInService(describeReq);
+        logger.info("SageMaker endpoint {} created: {}", endpointName, endpointArn);
+    }
+
+    /** Deletes the Amazon SageMaker endpoint. */
+    public void deleteEndpoint() {
+        logger.info("Deleting SageMaker endpoint {} ...", endpointName);
+        DeleteEndpointRequest req =
+                DeleteEndpointRequest.builder().endpointName(endpointName).build();
+        sageMaker.deleteEndpoint(req);
```

Comment:
Any cases this action might be failed?

Reply:
probably yes if the endpoint does not exist.

Reply:
same comment is applicable to next few functions.